### PR TITLE
Added wishlist and cart page

### PR DESCRIPTION
**Pull request by parthgabani777**
- Added wishlist and cart page
- changes in header and product/home card to display cart and wishlist information

### CODE REVIEW

**Review by nerdyspook — APPROVED** [review]
Code looks really clean

File: src/components/card/home-card.js [hunk 1]
```diff
@@ -14,7 +17,23 @@ function HomeCard({ product }) {
...
+                    {findItems(cart.itemInCart, product) ? (
+                        <button className="btn btn-primary">
+                            <Link to="/cart">Go To Cart</Link>
+                        </button>
+                    ) : (
+                        <button
+                            className="btn btn-secondary"
+                            onClick={() => {
+                                dispatchCart({
+                                    type: "ADD_TO_CART",
```
Comment (SanjayJatti) [inline src/components/card/home-card.js:29]: It will be better if created "Constants" for type

File: src/pages/wishlist/WishList.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    const navigation = useNavigate();
+
+    useEffect(() => {
+        auth.isAuthorized || navigation("/login");
+    }, []);
+
+    return (
+        <section className="main">
+            <h1 className="text-center text-l p-2">
+                My Wishlist ({wishlist.itemInWishlist.length})
```
Comment (SanjayJatti) [inline src/pages/wishlist/WishList.js]: You can further destructure '"temInWishlist" above return

File: src/pages/cart/Cart.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    useEffect(() => {
+        auth.isAuthorized || navigation("/login");
+    }, []);
+
+    const totalPrice = normalPrice - discountedPrice;
+
+    return (
+        <section className="main cart">
+            <h1 className="text-center text-m p-2">
+                My Cart({cart.itemInCart.length})
```
Comment (SanjayJatti) [inline src/pages/cart/Cart.js]: you can destructure "itemInCart"

**Review by SanjayJatti — APPROVED** [review]
- Code is Clean
Great job!

**Pull request merged**